com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/validation/ScopeJavaValidator.java

```diff
@@ -340,6 +342,23 @@ public void checkGuardDefaultExists(final ScopeRule context) {
     }
   }
 
+  /**
+   * Check that naming functions are used when referencing a feature type with no name feature.
+   *
+   * @param scope
+   *          the scope definition to check
+   */
+  @Check
+  public void checkNamingFunctionPresentNoNameFeature(final ScopeDefinition scope) {

```

Comment:
i think this validation doesn't cover the case when a naming function is defined in the 'naming' Btw, what was the use-case for this check?

Reply:
Just noticed this PR. My PR seems to be related: https://github.com/dsldevkit/dsl-devkit/pull/77.